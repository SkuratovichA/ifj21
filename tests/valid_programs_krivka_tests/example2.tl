-- Program 2: Vypocet faktorialu (rekurzivne)
require "ifj21"

function factorial(n : integer) : integer
  local n1 : integer = n - 1
  if n < 2 then
    return 1
  else
    local tmp : integer = factorial(n1)
    return n * tmp
  end
end

function main()
  write("Zadejte cislo pro vypocet faktorialu: ")
  local a : integer = readi()
  if a ~= nil then
		if a < 0 then
			write("Faktorial nejde spocitat!", "\n")
		else
			local vysl : integer = factorial(a)
			write("Vysledek je ", vysl, "\n")
    end
  else
     write("Chyba pri nacitani celeho cisla!\n")
  end
end

main()