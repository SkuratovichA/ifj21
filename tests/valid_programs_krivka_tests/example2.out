Zadejte cislo pro vypocet faktorialu: Vysledek je 720
